include/posix/time.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef __POSIX_TIME_H__
+#define __POSIX_TIME_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_NEWLIB_LIBC
+#include_next <time.h>
+#else
+struct timespec {
+	s32_t tv_sec;
+	s32_t tv_nsec;
+};
+#endif /* CONFIG_NEWLIB_LIBC */
+#include <kernel.h>
+#include <errno.h>
+#include "sys/types.h"
+
+#define CLOCK_MONOTONIC 1
+
+int clock_gettime(clockid_t clock_id, struct timespec *ts);
+
+/**
+ * @brief POSIX threading compatibility API
+ *
+ * See IEEE 1003.1
```

Comment:
Might be better to just point out to a [online copy of a man page](http://pubs.opengroup.org/onlinepubs/7908799/xsh/clock_settime.html) for `clock_settime()` instead.

Reply:
If linking to the man page, please don't forget to document the differences in Zephyr. In particular, for this API, document the clock_id type that is supported.